sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -466,14 +466,14 @@ object ScalaReflection extends ScalaReflection {
 
         case t if t <:< localTypeOf[Product] =>
           val params = getConstructorParameters(t)
-
-          CreateNamedStruct(params.flatMap { case (fieldName, fieldType) =>
+          val nullOutput = expressions.Literal.create(null, silentSchemaFor(t).dataType)
+          val nonNullOutput = CreateNamedStruct(params.flatMap { case (fieldName, fieldType) =>
```

Comment:
we can move `nullOutput` after `nonNullOutput`, so that we can use `nonNullOutput.dataType` instead of `silentSchemaFor(t).dataType`, which saves one computattion of schema.(not a big deal though)